### PR TITLE
Use new `datadog-ci` standalone binary 

### PR DESCRIPTION
**Pull request by juan-fernandez**
### What does this PR do?

Make it simpler to use `datadog-ci`: use new standalone binary instead of setting up node and installing `datadog-ci` with `npm`

### Review checklist

Please check relevant items below:

- [ ] This PR includes all [newly recorded cassettes](/DEVELOPMENT.md) for any modified tests.

- [x] This PR does not rely on API client schema changes.
  - [x] The CI should be fully passing.
- [ ] Or, this PR relies on API schema changes and this is a Draft PR to include tests for that new functionality.
  - Note: CI shouldn't be run on this Draft PR, as its expected to fail without the corresponding schema changes.

### CODE REVIEW

File: .github/workflows/test_integration.yml [hunk 1]
```diff
@@ -54,7 +54,9 @@ jobs:
         with:
           node-version: 14
       - name: Report source code metadata
-        run: npx @datadog/datadog-ci git-metadata upload
+        run: |
+          curl -L "https://github.com/DataDog/datadog-ci/releases/latest/download/datadog-ci" --output "/usr/bin/datadog-ci" && chmod +x "/usr/bin/datadog-ci"
```
Comment (jirikuncar) [inline .github/workflows/test_integration.yml:58]:
```suggestion
          curl -L "https://github.com/DataDog/datadog-ci/releases/latest/download/datadog-ci" --output "/usr/bin/datadog-ci"
          chmod +x "/usr/bin/datadog-ci"
```

**Review by jirikuncar — APPROVED** [review]

File: .github/workflows/test_integration.yml [hunk 2]
```diff
@@ -54,7 +54,11 @@ jobs:
         with:
           node-version: 14
       - name: Report source code metadata
-        run: npx @datadog/datadog-ci git-metadata upload
+        run: |
+          mkdir -p $GITHUB_WORKSPACE/bin
+          curl -L "https://github.com/DataDog/datadog-ci/releases/latest/download/datadog-ci" --output "$GITHUB_WORKSPACE/bin/datadog-ci"
+          chmod +x "$GITHUB_WORKSPACE/bin/datadog-ci"
+          $GITHUB_WORKSPACE/bin/datadog-ci git-metadata upload
```
Comment (jirikuncar) [inline .github/workflows/test_integration.yml:61]: @juan-fernandez do we need to build the package before uploading git metadata?

Comment (juan-fernandez) [reply .github/workflows/test_integration.yml:61]: what package? `datadog-ci`?

File: .github/workflows/test_integration.yml [hunk 3]
```diff
@@ -49,12 +49,11 @@ jobs:
           repo: datadog-api-spec
           status: pending
           context: integration
-      - name: Set up Node 14
-        uses: actions/setup-node@v2
-        with:
-          node-version: 14
```
Comment (jirikuncar) [inline .github/workflows/test_integration.yml:55]: we should probably keep it here

Comment (juan-fernandez) [reply .github/workflows/test_integration.yml]: added it back 👍

**Comment by juan-fernandez** [comment]
@jirikuncar good to merge?

**Pull request merged**